tools/apidiff/merger.cs

```diff
@@ -13,11 +15,14 @@ static string GetVersion (string line)
 		return line.Substring (start, end - start);
 	}
 
-	public static void Process (string platform, string path, string os)
+	public static void Process (string platform, string path, string os, string? destinationPath)
 	{
 		if (!Directory.Exists (path))
 			throw new DirectoryNotFoundException (path);
 
+		if (!Directory.Exists (destinationPath))
```

Comment:
I'm a bit confused on [this](https://github.com/xamarin/xamarin-macios/pull/13512/commits/69fb39023d282e8a83bfcf7d7628a56d035cf86c).

destinationPath is declared as a string?, so possibly null. What happens when you pass null to Exists....

```
csharp> using System.IO;
csharp> Directory.Exists(null)                                                                                                                                                                false  
```

Ok, that is unexpected, I'd have expected a throw ArgumentNull or something....

In any case, please re-add the `is not null` check, unless I'm missing something

Reply:
Oh wait yes you are right! If this merger.exe is ran with only 3 args, we want it to still be able to go through this function so we should keep the check for null in first!